src/loggers/logging.rs

```diff
@@ -60,6 +63,23 @@ where
     write_args(record, write)
 }
 
+#[cfg(feature = "chrono")]
+#[inline(always)]
+pub fn write_time<W>(write: &mut W, config: &Config) -> Result<(), Error>
+where
+    W: Write + Sized,
+{
```

Comment:
So here we would need to match the `time_format` instead and just hardcode `&'static str`-format strings for Rfc2822 and Rfc3339. chrono directly supports Rfc3339 with `%+`, but Rfc2822 would need to be constructed manually.